benchmark/worker/benchmark_client.go

```diff
@@ -273,28 +273,21 @@ func (bc *benchmarkClient) unaryLoop(conns []*grpc.ClientConn, rpcCountPerConn i
 				// The worker client needs to wait for some time after client is created,
 				// before starting benchmark.
 				if poissonLambda == nil { // Closed loop.
-					done := make(chan bool)
 					for {
-						go func() {
-							start := time.Now()
-							if err := benchmark.DoUnaryCall(client, reqSize, respSize); err != nil {
-								select {
-								case <-bc.stop:
-								case done <- false:
-								}
-								return
-							}
-							elapse := time.Since(start)
-							bc.lockingHistograms[idx].add(int64(elapse))
+						start := time.Now()
+						if err := benchmark.DoUnaryCall(client, reqSize, respSize); err != nil {
 							select {
 							case <-bc.stop:
-							case done <- true:
+								return
+							default:
 							}
-						}()
+							continue
+						}
+						elapse := time.Since(start)
+						bc.lockingHistograms[idx].add(int64(elapse))
 						select {
 						case <-bc.stop:
-							return
-						case <-done:
+						default:
 						}
```

Comment:
What does this `select` really do?

Reply:
There should have been a return in the `case <-bc.stop:` that got removed in the PR. Re-added it. The `bc.stop` channel allows the benchmark to be stopped gracefully.

Reply:
I've simplified this check now. No it only checks for the benchmark ending once, before making the RPC. In case of failures, the next iteration of the loop will perform the check and return.